### DISCUSSION TITLE
Running fine on RHEL 8 about half an hour then got "The LDAP server is unavailable." exception.

### DISCUSSION THREAD

**Discussion by caotoulei** [dotnet/aspnetcore#49088] [Q&A] [answered]
Here is the related code in my .net core API project Program.cs file.
...
builder.Services.AddAuthentication(NegotiateDefaults.AuthenticationScheme)
     .AddNegotiate(options =>
     {
         if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
         {
             options.EnableLdap(settings =>
             {
                 settings.Domain = "xxxxx.COM";
                 var ldapConnection = new LdapConnection(
                     new LdapDirectoryIdentifier("*ldap ip*", 389, true, false), 
                     new NetworkCredential(ApplicationData.srv_username, ApplicationData.srv_password), 
                     AuthType.Basic);
                 ldapConnection.SessionOptions.ReferralChasing = ReferralChasingOptions.None;
                 
                 ldapConnection.AutoBind = true;

                 settings.LdapConnection = ldapConnection;

             });
         }
     });
...
It runs fine on RHEL 8 Apache for about half an hour, and all the POST/GET requests to get responses right.  Then, all the requests throw exceptions like this:

_Jun 28 23:16:46 RHServer0 dotnet[56593]: Connection id "0HMRO6D462Q9A", Request id "0HMRO6D462Q9A:00000003": An unhandled exception was thrown by the application.
Jun 28 23:16:46 RHServer0 dotnet[56593]: System.DirectoryServices.Protocols.LdapException: The LDAP server is unavailable.
Jun 28 23:16:46 RHServer0 dotnet[56593]: at System.DirectoryServices.Protocols.LdapConnection.BeginSendRequest(DirectoryRequest request, TimeSpan requestTimeout, PartialResultProcessing partialMode, AsyncCa
Jun 28 23:16:46 RHServer0 dotnet[56593]: at System.DirectoryServices.Protocols.LdapConnection.BeginSendRequest(DirectoryRequest request, PartialResultProcessing partialMode, AsyncCallback callback, Object s
Jun 28 23:16:46 RHServer0 dotnet[56593]: at System.Threading.Tasks.TaskFactory`1.FromAsyncImpl[TArg1,TArg2](Func`5 beginMethod, Func`2 endFunction, Action`1 endAction, TArg1 arg1, TArg2 arg2, Object state,
Jun 28 23:16:46 RHServer0dotnet[56593]: at Microsoft.AspNetCore.Authentication.Negotiate.LdapAdapter.RetrieveClaimsAsync(LdapSettings settings, ClaimsIdentity identity, ILogger logger)
Jun 28 23:16:46 RHServer0 dotnet[56593]: at Microsoft.AspNetCore.Authentication.Negotiate.NegotiateHandler.HandleAuthenticateAsync()
Jun 28 23:16:46 RHServer0 dotnet[56593]: at Microsoft.AspNetCore.Authentication.AuthenticationHandler`1.AuthenticateAsync()
Jun 28 23:16:46 RHServer0 dotnet[56593]: at Microsoft.AspNetCore.Authentication.AuthenticationService.AuthenticateAsync(HttpContext context, String scheme)
Jun 28 23:16:46 RHServer0 dotnet[56593]: at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
Jun 28 23:16:46 RHServer0 dotnet[56593]: at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)_

And the Service is still Active. 
A lot of thanks for this.

**Comment by gfoidl** [comment, marked as answer]
Moved to an issue: https://github.com/dotnet/aspnetcore/issues/49105
Let's follow along over there.

**Reply by corsiva** [reply]
Hi @gfoidl ,
I am facing the same issue.
Is this issue fixed?
